server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -470,6 +470,25 @@ public boolean isRemoteStoreNode() {
         return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
     }
```

Comment:
Are there any consumers of `isRemoteStoreNode( )` after the addition of `isRemoteDataNode( )` and `isRemoteClusterNode( )`?

If not, we should remove this to avoid silent failures in the future. If there are any consumers, we can evaluate if it is possible to use the more specific methods that we are introducing.

Reply:
+1 to this